packages/fields/src/actions/set-as-homepage.tsx

```diff
@@ -0,0 +1,247 @@
+/**
+ * WordPress dependencies
+ */
+import { select, useDispatch } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+import type { Settings, Page } from '@wordpress/core-data';
+import { __, sprintf } from '@wordpress/i18n';
+import {
+	Button,
+	__experimentalText as Text,
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+} from '@wordpress/components';
+import type { Action, ActionModal } from '@wordpress/dataviews';
+import { store as noticesStore } from '@wordpress/notices';
+
+/**
+ * Internal dependencies
+ */
+import { getItemTitle } from './utils';
+import type { CoreDataError, PostWithPermissions } from '../types';
+
+const PAGE_POST_TYPE = 'page';
+
+const SetAsHomepageModal: ActionModal< PostWithPermissions >[ 'RenderModal' ] =
+	( { items, closeModal, onActionPerformed } ) => {
+		const [ item ] = items;
+		const pageTitle = getItemTitle( item );
+		const siteSettings = select( coreStore ).getEntityRecord< Settings >(
```

Comment:
You should use `useSelect` here.

Reply:
Swapped this for `useSelect` in https://github.com/WordPress/gutenberg/pull/65426/commits/7687e3c43d81578ea184e4e2bb73bb6d4fd6e73d.